app/Models/Building.php

```diff
@@ -126,6 +127,41 @@ public function getTagsAttribute()
         return $tags;
     }
 
+    public function getUrlAttribute()
+    {
+        $url = route('building.detail', [$this->id, $this->slug]); // @todo: id or source_id?
+        return $url;
+        // return \LaravelLocalization::getNonLocalizedURL($url);
```

Comment:
Definitely `id`.  `source_id` maps the record to the Access database. (Perhaps it should've been called `upstream_id`)

```suggestion
        return route('building.detail', [$this->id, $this->slug]);
```
If we make it this way, I wonder if there's any need for this attribute after all?